scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TerminationData.java

```diff
@@ -134,4 +148,42 @@ public void run() {
             service.terminateJobHandling(jobId);
         }
     }
+
+    @NotNull
+    public Map<String, Serializable> getStringSerializableMap(SchedulingService service, TaskTerminationData taskToTerminate) throws IOException, ClassNotFoundException {
+        Map<String, Serializable> variables = new HashMap<>();
+
+        RunningTaskData taskData = taskToTerminate.taskData;
+        TaskResultImpl taskResult = taskToTerminate.taskResult;
+        InternalJob internalJob = taskToTerminate.internalJob;
+
+        if (!taskToTerminate.normalTermination || taskResult == null) {
+            List<InternalTask> iDependences = taskData.getTask().getIDependences();
+            if (iDependences != null) {
+                List<TaskId> parentIds = new ArrayList<>(iDependences.size());
+                for (int i = 0; i < iDependences.size(); i++) {
+                    parentIds.add(iDependences.get(i).getId());
+                }
+                Map<TaskId, TaskResult> taskResults = service.getInfrastructure().getDBManager()
+                        .loadTasksResults(
+                                taskData.getTask().getJobId(), parentIds);
+                for (TaskResult currentTaskResult : taskResults.values()) {
+                    if (currentTaskResult.getPropagatedVariables() != null) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)